### ISSUE TITLE
type error in v3.1.0

### ISSUE THREAD

**Issue by sdg9670** [sidorares/node-mysql2#1816]
I couldn't find connection.promise()

```typescript
mysql.createConnection().promise()
```

**Comment by fated-x** [comment]
I too am experiencing this issue. The root cause appears to be the fact that `/node_modules/mysql2/typings/mysql/lib/Pool.d.ts` declared `Pool` class has a promise property that returns an `any` type:

```
Line 74: promise(promiseImpl?: any): any;
```

**Comment by anders8** [comment]
I originally experienced a similar issue with 3.0.0 and like @fated-x I reverted, to the 2.2.5 I was using in my case.  I just tried 3.2.0 and it's compiling without errors and making connections without apparent trouble.

However, I'm using mysql.createPool() not mysql.createConnection()

**Comment by marcellino-ornelas** [comment]
Im using `3.2.0` and im getting a error when trying to use promise on connection

`Property 'promise' does not exist on type 'Connection'.ts(2339)`

```
import { createConnection, Connection } from 'mysql2';
import { config } from '@src/config';

export const db: Connection = createConnection(config.db);

db.promise(); // <---- error
```

**Comment by tristan-nsg** [comment]
I'm still getting this issue on 3.3.0

**Comment by wellwelwel** [comment]
@tristan-nsg

Look at this fix:
- #1995

This fix is relatively extensive, so its review cannot be rushed 🙋🏻‍♂️

**Comment by wellwelwel** [comment]
> @sdg9670, sorry for the delay

This issue has been fixed in `3.3.1` version.

Can you check it again in `^3.4.0`?

**Comment by wellwelwel** [comment]
After version `^3.3.5`, you can follow this example:
https://github.com/sidorares/node-mysql2/blob/00508c38a13f1394fb8a6cf3fd88df1353ba0e77/test/tsc-build/mysql/createConnection/promise/execute.ts#L5-L11

---

I'm closing this **Issue**, but feel free to ask anything 🙋🏻‍♂️
